src/_nebari/upgrade.py

```diff
@@ -1286,13 +1302,32 @@ def _version_specific_upgrade(
                 username="root",
                 password=config["security"]["keycloak"]["initial_root_password"],
             )
-
-            # Proceed with updating group permissions
+            # Get client ID as role is bound to the JupyterHub client
             client_id = keycloak_admin.get_client_id("jupyterhub")
-            role_name = "allow-group-directory-creation-role"
+            role_name = "legacy-group-directory-creation-role"
+
+            # Create the role if it doesn't exist
+            # If the role does not exist, create it
```

Comment:
Seems like one of these comments could be removed

Reply:
tks